### PR TITLE
Normative Optional: Add WeakRef.prototype.constructor

### LINKED ISSUE #131: Add WeakRef.prototype.constructor
the WeakRef.prototype.constructor is missing in the proposed specs, add it for consistency with other built-in constructors?

### PR DESCRIPTION
**Pull request by littledan**
Fixes #131

### CODE REVIEW

**Comment by littledan** [comment]
cc @tschneidereit @erights @leobalter

File: spec/weak-ref.html [hunk 1]
```diff
@@ -89,6 +89,18 @@ <h1>Properties of the WeakRef Prototype Object</h1>
...
 
+    <emu-clause id="sec-weak-ref.prototype.constructor" normative-optional>
+      <h1>WeakRef.prototype.constructor</h1>
+
+      <p>The initial value of `WeakRef.prototype.constructor` is the intrinsic object %WeakRef%.</p>
+
+      <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</p>
+
+      <emu-note type="editor">
+        This section is to be treated identically to the "Annex B" of ECMA-262, but to be written in-line with the main specification.
```
Comment (ljharb) [inline spec/weak-ref.html]: can you elaborate on this conclusion?

File: spec/index.html [hunk 2]
```diff
@@ -8,6 +8,19 @@
 copyright: proposal
 contributors: Dean Tribble, Till Schneidereit, Sathya Gunasekaran
 </pre>
+<style>
+[normative-optional] {
```
Comment (erights) [inline spec/index.html]: Could you send me a screenshot of how the normal optional text below gets rendered with these stylings?

File: spec/weak-ref.html (hunk 1 above)
Comment (littledan) [reply spec/weak-ref.html]: The idea here is to put Annex B inline to make it easier to read. We do this in ECMA-402. We discussed this earlier today in the TC39 meeting; delegates can see the draft notes. Do you have any concerns with this idea?

Comment (ljharb) [reply spec/weak-ref.html]: It seems fine, but that's a pretty massive editorial change discussed with no editors present, and wasn't what I expected from a slideless presentation called "reform".

Reviewing the notes gives me a bit more clarity, but I'd still hope to see a few examples of inline normative optional land in the main spec before a proposal added one (that shouldn't block this PR ofc)

Comment (littledan) [reply spec/weak-ref.html]: Yeah, the plan is that I'll file an issue in ecma262 describing the plan, and we'll work from there in PRs. This PR can be thought of as a really early first draft, editorially; we just need to get it off the ground here and have the observable semantics specified.

File: spec/index.html (hunk 2 above)
Comment (littledan) [reply spec/index.html]: image

Comment (erights) [reply spec/index.html]: Excellent! I am all in favor. Thanks.

**Comment by littledan** [comment]
OK, I'm landing this PR, since it's an important semantic to get right by Stage 3, and it's had some review here.

**Pull request merged**